integrations/opennms-wsman-asset-provisioning-adapter/src/main/java/org/opennms/netmgt/provision/WsManAssetProvisioningAdapter.java

```diff
@@ -0,0 +1,463 @@
+         */
+        public void setEventForwarder(final EventForwarder eventForwarder) {
+                m_eventForwarder = eventForwarder;
+        }
+
+        /**
+         * @return the m_config
+         */
+        public WsManAssetAdapterConfig getWsManAssetAdapterConfig() {
+                return m_config;
+        }
+
+        /**
+         * @param mConfig the m_config to set
+         */
+        public void setWsManAssetAdapterConfig(final WsManAssetAdapterConfig mConfig) {
+                m_config = mConfig;
+        }
+
+        private static String nodeToString(Node node) {
+                NodeList children = node.getChildNodes();
+                Node child = children.item(0);
+
+                return child.getTextContent();
+        }
+
+        /**
+         * <p>getName</p>
+         *
+         * @return a {@link java.lang.String} object.
+         */
+        @Override
+        public String getName() {
+                return NAME;
+        }
+
+        private InetAddress getIpForNode(final OnmsNode node) {
+                LOG.debug("getIpForNode: node: {} Foreign Source: {}", node.getNodeId(), node.getForeignSource());
+                final OnmsIpInterface primaryInterface = node.getPrimaryInterface();
+                InetAddress ipaddr = InetAddressUtils.getLocalHostAddress();
+                if (primaryInterface == null) {
+                        LOG.debug("getIpForNode: found null SNMP Primary Interface, getting interfaces");
+                        final Set<OnmsIpInterface> ipInterfaces = node.getIpInterfaces();
+                        for (final OnmsIpInterface onmsIpInterface : ipInterfaces) {
+                                LOG.debug("getIpForNode: trying Interface with id: {}", onmsIpInterface.getId());
+                                if (InetAddressUtils.str(onmsIpInterface.getIpAddress()) != null)
+                                        ipaddr = onmsIpInterface.getIpAddress();
+                                else
+                                        LOG.debug("getIpForNode: found null ip address on Interface with id: {}", onmsIpInterface.getId());
+
+
+                        }
+                } else {
+                        LOG.debug("getIpForNode: found SNMP Primary Interface");
+                        if (InetAddressUtils.str(primaryInterface.getIpAddress()) != null )
+                                ipaddr = primaryInterface.getIpAddress();
+                        else
+                                LOG.debug("getIpForNode: found null ip address on Primary Interface");
+                }
+                return ipaddr;
+        }
+
+        /**
+         * <p>handleReloadConfigEvent</p>
+         *
+         * @param event a {@link org.opennms.netmgt.xml.event.Event} object.
+         */
+        @EventHandler(uei = EventConstants.RELOAD_DAEMON_CONFIG_UEI)
+        public void handleReloadConfigEvent(final Event event) {
+                if (isReloadConfigEventTarget(event)) {
```

Comment:
Can eliminate this boilerplate by using `org.opennms.netmgt.daemon.DaemonTools.handleReloadEvent()` instead.

Reply:
Something like:
`import org.opennms.netmgt.daemon.DaemonTools;
...
        @EventHandler(uei = EventConstants.RELOAD_DAEMON_CONFIG_UEI)
        public void handleReloadEvent(Event e) {
            DaemonTools.handleReloadEvent(e, Provisiond.NAME, (event) -> handleConfigurationChanged());
        }
`

?